cli/cmd/project/splits.go

```diff
@@ -0,0 +1,107 @@
+package project
+
+import (
+	"fmt"
+
+	"github.com/rilldata/rill/cli/pkg/cmdutil"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	runtimeclient "github.com/rilldata/rill/runtime/client"
+	"github.com/spf13/cobra"
+)
+
+func SplitsCmd(ch *cmdutil.Helper) *cobra.Command {
+	var project, path, model string
+	var local bool
+	var pageSize uint32
+	var pageToken string
+
+	splitsCmd := &cobra.Command{
+		Use:   "splits [<project>] <model>",
+		Args:  cobra.MaximumNArgs(2),
+		Short: "List splits for a model",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := ch.Client()
+			if err != nil {
+				return err
+			}
+
+			if len(args) == 1 {
+				model = args[0]
+			} else if len(args) == 2 {
+				project = args[0]
+				model = args[1]
+			}
+
+			if !local && !cmd.Flags().Changed("project") && len(args) <= 1 && ch.Interactive {
+				project, err = ch.InferProjectName(cmd.Context(), ch.Org, path)
+				if err != nil {
+					return err
+				}
+			}
+
+			var host, instanceID, jwt string
+			if local {
+				// This is the default port that Rill localhost uses for gRPC.
+				// TODO: In the future, we should capture the gRPC port in ~/.rill and use it here.
+				host = "http://localhost:49009"
+				instanceID = "default"
+			} else {
+				proj, err := client.GetProject(cmd.Context(), &adminv1.GetProjectRequest{
+					OrganizationName: ch.Org,
+					Name:             project,
+				})
+				if err != nil {
+					return err
+				}
+
+				depl := proj.ProdDeployment
+				if depl == nil {
+					return fmt.Errorf("project %q is not currently deployed", project)
+				}
+				if depl.Status != adminv1.DeploymentStatus_DEPLOYMENT_STATUS_OK {
+					ch.PrintfWarn("Deployment status not OK: %s\n", depl.Status.String())
+					return nil
+				}
+
+				host = depl.RuntimeHost
+				instanceID = depl.RuntimeInstanceId
+				jwt = proj.Jwt
+			}
+
+			rt, err := runtimeclient.New(host, jwt)
+			if err != nil {
+				return fmt.Errorf("failed to connect to runtime: %w", err)
+			}
+
+			res, err := rt.GetModelSplits(cmd.Context(), &runtimev1.GetModelSplitsRequest{
```

Comment:
It actually primarily orders and paginates using the index of the split in the resolver result, not the hash. So this means it will show splits in the order they were returned from the `splits` resolver, which is also the order they will be scheduled in.

I think this is the expected default ordering, but I agree it would be nice to be able to customize the ordering and also add filters to only show pending and errored splits. I will tackle that in a follow up PR, this PR is already doing a lot :)